support-workers/elasticsearch/cloudWatchToElasticSearchLambda.js

```diff
@@ -0,0 +1,261 @@
+  }
+
+  return {};
+}
+
+function extractJson(message) {
+  var jsonStart = message.indexOf('{');
+  if (jsonStart < 0) return null;
+  var jsonSubString = message.substring(jsonStart);
+  return isValidJson(jsonSubString) ? jsonSubString : null;
+}
+
+function isValidJson(message) {
+  try {
+    JSON.parse(message);
+  } catch (e) { return false; }
+  return true;
+}
+
+function isNumeric(n) {
+  return !isNaN(parseFloat(n)) && isFinite(n);
+}
+
+function post(body, callback) {
+  var requestParams = buildRequest(endpoint, body);
+
+  var request = https.request(requestParams, function(response) {
+    var responseBody = '';
+    response.on('data', function(chunk) {
+      responseBody += chunk;
+    });
+
+    response.on('end', function() {
+      var info = JSON.parse(responseBody);
+      var failedItems;
+      var success;
+      var error;
+
+      if (response.statusCode >= 200 && response.statusCode < 299) {
+        failedItems = info.items.filter(function(x) {
+          return x.index.status >= 300;
+        });
+
+        success = {
+          "attemptedItems": info.items.length,
+          "successfulItems": info.items.length - failedItems.length,
+          "failedItems": failedItems.length
+        };
+      }
+
+      if (response.statusCode !== 200 || info.errors === true) {
+        // prevents logging of failed entries, but allows logging
+        // of other errors such as access restrictions
+        delete info.items;
+        error = {
+          statusCode: response.statusCode,
+          responseBody: info
+        };
+      }
+
+      callback(error, success, response.statusCode, failedItems);
+    });
+  }).on('error', function(e) {
+    callback(e);
+  });
+  request.end(requestParams.body);
+}
+
+function buildRequest(endpoint, body) {
+  var endpointParts = endpoint.match(/^([^\.]+)\.?([^\.]*)\.?([^\.]*)\.amazonaws\.com$/);
+  var region = endpointParts[2];
+  var service = endpointParts[3];
+  var datetime = (new Date()).toISOString().replace(/[:\-]|\.\d{3}/g, '');
+  var date = datetime.substr(0, 8);
+  var kDate = hmac('AWS4' + process.env.AWS_SECRET_ACCESS_KEY, date);
+  var kRegion = hmac(kDate, region);
+  var kService = hmac(kRegion, service);
+  var kSigning = hmac(kService, 'aws4_request');
+
+  var request = {
+    host: endpoint,
+    method: 'POST',
+    path: '/_bulk',
+    body: body,
+    headers: {
+      'Content-Type': 'application/json',
+      'Host': endpoint,
+      'Content-Length': Buffer.byteLength(body),
+      'X-Amz-Security-Token': process.env.AWS_SESSION_TOKEN,
+      'X-Amz-Date': datetime
+    }
+  };
+
+  var canonicalHeaders = Object.keys(request.headers)
+  .sort(function(a, b) { return a.toLowerCase() < b.toLowerCase() ? -1 : 1; })
+  .map(function(k) { return k.toLowerCase() + ':' + request.headers[k]; })
+  .join('\n');
+
+  var signedHeaders = Object.keys(request.headers)
+  .map(function(k) { return k.toLowerCase(); })
+  .sort()
+  .join(';');
```

Comment:
is it just me or is the indentation a bit off on these vars?

Reply:
also can we make any/all of the `var`s `const`s? Presumably we're running on a later version of node.